components/calendar/index.js

```diff
@@ -64,60 +58,70 @@ const Calendar = React.createClass({
 
   buildEvents: function (events) {
     let id = 0;
-    let lastEpochDays = 0;
     let daysEvents = [];
+    let previousEventDate;
 
     let preDays = [];
     let postDays = [];
     let preDaysSection = '';
 
     events.forEach(function (event, index) {
```

Comment:
Maybe a bit out of scope for this pr, but I think this function should be split into smaller functions. Both to reduce duplicate code and to make it clearer what each section actually does.